auraed/src/logging/log_channel.rs

```diff
@@ -47,11 +47,17 @@ pub struct LogChannel {
 
 impl LogChannel {
     /// Constructor creating the channel for log communication
-    pub fn new(name: &str) -> LogChannel {
+    pub fn new(name: String) -> LogChannel {
         // TODO: decide for a cap. 40 is arbitrary
         let (producer, _) = broadcast::channel(40);
-        LogChannel { producer, name: name.to_string() }
+        LogChannel { producer, name }
     }
+
+    /// Getter for the name
+    pub fn name(&self) -> &String {
```

Comment:
This should be &str. I don't think &String is ever really used. 

*pulling down the pr to review the rest*